sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Util.scala

```diff
@@ -329,4 +330,18 @@ private[sql] object CatalogV2Util {
       .getOrElse(catalogManager.v2SessionCatalog)
       .asTableCatalog
   }
+
+  def failCharType(dt: DataType): Unit = {
+    if (HiveStringType.containsCharType(dt)) {
+      throw new AnalysisException("Cannot use CHAR/VARCHAR type in non-Hive tables.")
```

Comment:
Shall we have a directional warning like `Use STRING type instead of CHAR/VARCHAR type in non-Hive tables.`?